### PR TITLE
ZynAddSubFx filter frequency set to 127

### PR DESCRIPTION
**Pull request by bratpeki**
I'm tying this PR to the [ZynAddSubFX PR #23](https://github.com/LMMS/zynaddsubfx/pull/23). These two PRs aim to disable the default lowpass filter applied to Zyn's three synths, as well as keeping the possibility of filter automation without affecting Zyn internally.

There is only one change in this PR specifically, that being that the filter frequency is set to the maximum value of 127, as per [Lost Robot's suggestion in the Discord server](https://discord.com/channels/203559236729438208/332258319228207114/1261735891068653609).

image

### CODE REVIEW

**Comment by super-manis** [comment]
Would it be, by any chance, nicer to set the default wave to Power instead of Sine?

**Review by LostRobotMusic — APPROVED** [review]

**Comment by bratpeki** [comment]
I've noticed that the plugin still treats the filter frequency knob as if it were set to 64. Only when the knob is moved does the change happen. Can you check this, @LostRobotMusic?

**Comment by bratpeki** [comment]
Here a [link to an MMPZ file](https://drive.google.com/file/d/11KJftT-au5LRnnoTld9RzQ2_eGVAYIVv/view?usp=sharing) proving that this even stays after saving. Will investigate further.

**Comment by bratpeki** [comment]
I've saved the MMPZ as an MMP and am reading it.

The first XML entry after each `<instrument name="zynaddsubfx">`, for all three instances of Zyn, reads the following:

```
freq set to 64:
<zynaddsubfx portamento="0" resbandwidth="64" bandwidth="64" filterfreq="64" modifiedcontrollers="74," filterq="64" rescenterfreq="64" fmgain="127" forwardmidicc="1">

freq set to 127:
<zynaddsubfx portamento="0" resbandwidth="64" bandwidth="64" filterfreq="127" modifiedcontrollers="74," filterq="64" rescenterfreq="64" fmgain="127" forwardmidicc="1">

knob not touched:
<zynaddsubfx portamento="0" resbandwidth="64" bandwidth="64" filterfreq="127" modifiedcontrollers="" filterq="64" rescenterfreq="64" fmgain="127" forwardmidicc="1">
```

The first two have had their "Filter frequency" knobs moved (indicated by `modifiedcontrollers="74,"`).

**Comment by bratpeki** [comment]
`ZynAddSubFx.cpp`, line 123:
```cpp
connect( &m_filterFreqModel, SIGNAL( dataChanged() ), this, SLOT( updateFilterFreq() ), Qt::DirectConnection );
```

This would explain why a change to the knob pointed to by `m_filterFreqModel` would result in the update of the filter frequency.

`updateFilterFreq()` is defined in the same file, on line 418:

```cpp
GEN_CC_SLOT(updateFilterFreq,C_filtercutoff,m_filterFreqModel);
```

The `GEN_CC_SLOT` macro expands to:

```cpp
void ZynAddSubFxInstrument::updateFilterFreq() {														 
    sendControlChange( C_filtercutoff, m_filterFreqModel.value() );
    m_modifiedControllers[C_filtercutoff] = true;
}
```

A hacky approach would be to call `updateFilterFreq()` from the initialization process of the plugin, although it should be possible to find the root cause of the issue and avoid that call.

**Comment by bratpeki** [comment]
Indeed, in the same file, in `ZynAddSubFxInstrument::ZynAddSubFxInstrument`, adding `updateFilterFreq();` to the end fixes the issue.

**Comment by bratpeki** [comment]
Found the root! It's in `plugins/ZynAddSubFx/zynaddsubfx/src/Params/Controller.cpp` under `Controller::defaults()` and `Controller::resetall()`. I'll update my ZynAddSubFx PR accordingly.

**Comment by JohannesLorenz** [comment]
Didn't you want to upgrade the submodule? You can reference it right now, assuming we will likely do an FF-merge in the zyn repo.

**Comment by bratpeki** [comment]
Could you point me to some resource regarding that?

**Comment by bratpeki** [comment]
After trying, I was unable to update the LMMS zynaddsubfx submodule. Is it possible for someone more experienced to assist me with this?

**Comment by Rossmaxx** [comment]
IMO, you should wait till the other PR is merge.

**Comment by bratpeki** [comment]
Likewise, zynaddsubfx first, then update the submodule, then merge this.

**Comment by JohannesLorenz** [comment]
This should work (if not we can still fix later):

1. Go to the submodule
2. Checkout the version that you want to use (I assume that the submodule has already been pushed)
3. Go back to lmms
4. git add the submodule
5. git commit
6. git push

**Comment by bratpeki** [comment]
My fork of LMMS is now pointing to the master commit of my fork of Zyn!

**Comment by JohannesLorenz** [comment]
> I ended up using `git submodule set-url plugins/ZynAddSubFx/zynaddsubfx https://github.com/bratpeki/zynaddsubfx`. This has updated `.gitmodules`. Is this what you had in mind? If not, I will make further changes.

Sorry, I forgot that this requires switching the forks. It is not good to set the URL of the submodule to your fork, because in the end, it should point to ours. I now pushed branch `filterfreq` to our fork, with the content of your branch. If you manage to link to that from this branch (probably this requires you again to `set-url` back to our fork), then it should be alright...

**Comment by bratpeki** [comment]
That should be it! [Here's a link to the tree, to make things easier](https://github.com/bratpeki/lmms/tree/master/plugins/ZynAddSubFx)!

File: .gitmodules
Comment (JohannesLorenz) [inline .gitmodules:6]: The URL should not have changed at all, right?

Comment (bratpeki) [reply .gitmodules:6]: It should have, I set it to `https://github.com/bratpeki/zynaddsubfx.git` previously:

> I ended up using `git submodule set-url plugins/ZynAddSubFx/zynaddsubfx https://github.com/bratpeki/zynaddsubfx`. This has updated `.gitmodules`. Is this what you had in mind? If not, I will make further changes.

Comment (JohannesLorenz) [reply .gitmodules:6]: I mean, compared to master this should not have changed. You can `git reset master -- .gitmodules` and then commit/push.

Comment (bratpeki) [reply .gitmodules:6]: On it!

Comment (bratpeki) [reply .gitmodules:6]:
```
[peki@~/Code/lmms]$ git reset master -- .gitmodules
[peki@~/Code/lmms]$ git add .
[peki@~/Code/lmms]$ git diff
[peki@~/Code/lmms]$ git commit -m "Ran the command"
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
```

Seems as if, due to the command not changing anything now, it doesn't let the commit be made.

Comment (bratpeki) [reply .gitmodules:6]: Changing it manually right now

Comment (JohannesLorenz) [reply .gitmodules]: Manually is also good :+1:

**Comment by JohannesLorenz** [comment]
This PR seems buggy. I do this:

1. Open LMMS
2. Add a Zyn instance to the instruments
3. Open Zyn UI, go to global AD synth, look at Filterfreq

Observations:

1. The knob is not at maximum (as the PR suggests?)
2. The knob does not behave linear, but rather logarithmic (or exponential?)

Can this be confirmed? Or did I do anything wrong? At least, the behavior is correct on branch `lv2-ui`, which is very close to `master`.

**Comment by bratpeki** [comment]
Your statements are correct, the change in the frequency relating to LMMS is only in regard to the FREQ knob in the Zyn LMMS UI (red box in the image at the top). Lost checked that 100 is the best value for the ADDSynth, SUBSynth and PADSynth filter, as it makes sense in the context of automation as well as not making the signal lowpassed, so I set them to 100. The change that should be noticed is that none of the synths are lowpassed, as was the case previously.

The change to ADD-, SUB- and PADSynth is in the zynaddsubfx repo, of course.

**Comment by JohannesLorenz** [comment]
I don't understand. Can you reproduce my issue? If yes, IMO this should be fixed. I can't imagine that Lost planned that?

**Comment by bratpeki** [comment]
I'll try to explain.

By default, Zyn's synths are lowpassed, with Add having a filterfreq vaule of 94, IIRC. Understandably, having the signal lowpassed implies that the signal has been edited from the standard sine wave before you did anything. I aimed to correct that. Initially, I set all the filterfreq's to 127, but Lost pointed out that that would not allow for neatly automating the signal, as made evident by these Discord messages:

> Lost Robot — 07/13/2024 7:24 PM
> Approximately values 80-127 from LMMS's FREQ knob will have no impact by default if ZynAddSubFX's internal knob value is set to 127 by default.

> Lost Robot — 07/13/2024 7:27 PM
> Yes.  I recommend all three of those having a default value of 100 and LMMS's FREQ knob having a default value of 127.

I did as Lost suggested, which is what the two PRs I made are.

Tres made it clear that this is changed in Zyn-Fusion, so this is simply a small patch to make Zyn more user-friendly until it's is updated.

**Comment by bratpeki** [comment]
This should now really be it! Both here and in the Zyn repo.

**Comment by bratpeki** [comment]
Same as in the other repo, I am terribly sorry for misunderstanding your comment.
The commit message is:

```
Made it so ZynAddSubFx isn't lowpassed by default.

In this repo, the only changes made are updating the according submodule
and changing the FREQ knob in ZynAddSubFx's LMMS GUI.
```

**Comment by bratpeki** [comment]
After the Zyn PR is merged, the submodule here has to be updated, after which this can be merged as well.

**Comment by JohannesLorenz** [comment]
OK, please update the submodule reference now.

**Comment by JohannesLorenz** [comment]
I think that went wrong? I guess it should point to 9499523f70322b6034673566898300e7543fdf3e now, because this is the repo's master?

**Comment by bratpeki** [comment]
I believe it does, here's the [tree of my fork](https://github.com/bratpeki/lmms/tree/master/plugins/ZynAddSubFx).

**Comment by JohannesLorenz** [comment]
Now I see it, thanks.

**Pull request merged**
